features/cellular/framework/AT/ATHandler.cpp

```diff
@@ -295,12 +310,15 @@ void ATHandler::restore_at_timeout()
 
 void ATHandler::process_oob()
 {
+    ScopedLock<ATHandler> lock(*this);
+    _oob_pending = true;
+    _oob_queued = false;
```

Comment:
Not following the need for separate `pending` and `queued` flags. Don't we hold the `ATHandler` lock for the entire `process_oob`? So we're effectively atomically processing and clearing the `queued` flag? No-one can ever see `pending`? (Assuming they're accessing it within the lock only).

Reply:
As cancel is void function we decided not to call it. 
Replaced _oob_queued  and _oob_pending with _event_id.
Force pushed changes.